src/libstd/sync/mod.rs

```diff
@@ -17,17 +17,28 @@
 
 #![stable(feature = "rust1", since = "1.0.0")]
 
+#[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
>I'm not sure we've hammered out a good set of semantics for them to follow.

It can be decided later, when these annotations will actually be used by the checking pass. (I plan to look at the checking pass when I finish with the annotation pass)

I'd expect the first to be stable and the second to be unstable.
As a practical example - most of libcore items would be unstable, but their reexports from std would be stable. Now libcore is full of stable items because stability on reexports doesn't work.

Note, that many reexports currently have stability attributes, they are just not enforced.